library/MiscUtils.cpp

```diff
@@ -380,8 +418,30 @@ std::string UTF2DF(const std::string &in)
     return out;
 }
 
+DFHACK_EXPORT std::string UTF2CONSOLE(const std::string &in)
+{
+#ifdef WIN32
+    return UTF16ToCurrentCodePage(UTF8ToUTF16(in));
+#else
+    return in;
+#endif
+}
+
+DFHACK_EXPORT std::string UTF2CONSOLE(DFHack::color_ostream &out, const std::string &in)
+{
+    return out.is_console() ? UTF2CONSOLE(in) : in;
+}
+
 DFHACK_EXPORT std::string DF2CONSOLE(const std::string &in)
 {
+#ifdef WIN32
+    // convert to UTF-16
+    std::wstring win(in.size(), 0);
+    for (unsigned i = 0; i < in.size(); i++)
+        win[i] = character_table[in[i]];
```

Comment:
`in[i]` needs to be cast to `uint8_t`, otherwise extended characters will be negative.